lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1297,8 +1284,7 @@ OpFoldResult HeadPrimOp::fold(ArrayRef<Attribute> operands) {
     if (auto cst = getConstant(operands[0])) {
       int shiftAmount =
           input().getType().cast<IntType>().getWidthOrSentinel() - amount();
-      return getIntAttr(getType(),
-                        cst->lshr(shiftAmount).truncOrSelf(amount()));
+      return getIntAttr(getType(), cst->lshr(shiftAmount).trunc(amount()));
```

Comment:
I' wanted this line and the `shiftAmount` above to be reduced to:
```
cst->extractBits(amount(), cst->getBitWidth() - amount())
```
But it doesn't work the same (asserts out or maybe worse) when asking for, say, top zero bits.  Oh well :).